lib/mutgen.c

```diff
@@ -1221,7 +1221,8 @@ mutgen_choose_alleles(mutgen_t *self, tsk_id_t *parent, mutation_t **bottom_muta
         } else {
             parent_mut = bottom_mutation[u];
             mut->parent = parent_mut;
-            if (mut->time > parent_mut->time || (parent_mut->new && !mut->new)) {
+            if (mut->time > parent_mut->time
+                || (!allow_ancestral && (parent_mut->new && !mut->new))) {
```

Comment:
I'm not sure about this check - I think that it should be impossible for mut->time to be greater than parent_mut->time, at least now, if not before. So maybe this should be
```
assert(mut-.time <= parent_mut->time);
if (!allow_ancestral && (parent_mut->new && !mut->new)) {
...
```
... ?
```

Reply:
agree. I'll change it